src/Aspire.Hosting.Nats/NatsBuilderExtensions.cs

```diff
@@ -137,6 +139,155 @@ public static IResourceBuilder<NatsServerResource> WithJetStream(this IResourceB
         return builder.WithArgs("-js");
     }
 
+    /// <summary>
+    /// Configures a container resource for Nui (a web UI for NATS) which is pre-configured to connect to the <see cref="NatsServerResource"/> that this method is used on.
+    /// </summary>
+    /// <param name="builder">The NATS server resource builder.</param>
+    /// <param name="configureContainer">Configuration callback for Nui container resource.</param>
+    /// <param name="containerName">The name of the container (Optional).</param>
+    /// <example>
+    /// Use in application host with a NATS resource
+    /// <code lang="csharp">
+    /// var builder = DistributedApplication.CreateBuilder(args);
+    ///
+    /// var nats = builder.AddNats("nats")
+    ///    .WithJetStream()
+    ///    .WithNui();
+    ///
+    /// var api = builder.AddProject&lt;Projects.Api&gt;("api")
+    ///   .WithReference(nats);
+    ///
+    /// builder.Build().Run();
+    /// </code>
+    /// </example>
+    /// <remarks>
+    /// This version of the package defaults to the <inheritdoc cref="NatsContainerImageTags.NuiTag"/> tag of the <inheritdoc cref="NatsContainerImageTags.NuiImage"/> container image.
+    /// </remarks>
+    /// <returns>A reference to the <see cref="IResourceBuilder{T}"/>.</returns>
+    public static IResourceBuilder<NatsServerResource> WithNui(this IResourceBuilder<NatsServerResource> builder,
+        Action<IResourceBuilder<NuiResource>>? configureContainer = null,
+        string? containerName = null)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        if (builder.ApplicationBuilder!.Resources.OfType<NuiResource>().SingleOrDefault() is { } existingNuiResource)
+        {
+            var builderForExistingResource = builder.ApplicationBuilder.CreateResourceBuilder(existingNuiResource);
+            configureContainer?.Invoke(builderForExistingResource);
+            return builder;
+        }
+        else
+        {
+            containerName ??= $"{builder.Resource.Name}-nui";
+
+            builder.ApplicationBuilder.Services.AddHttpClient();
+
+            var nuiContainer = new NuiResource(containerName);
+            var nuiContainerBuilder = builder.ApplicationBuilder.AddResource(nuiContainer)
+                                                 .WithImage(NatsContainerImageTags.NuiImage, NatsContainerImageTags.NuiTag)
+                                                 .WithImageRegistry(NatsContainerImageTags.NuiRegistry)
+                                                 .WithHttpEndpoint(targetPort: 31311, name: "http")
+                                                 .WithVolume(VolumeNameGenerator.Generate(builder, "db"), "/db")
```

Comment:
Typically we make this "opt-in" with a `.WithDataVolume` extension method on the `IResourceBuilder<NuiResource> builder`.